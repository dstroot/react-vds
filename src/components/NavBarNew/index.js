import React, { useState } from 'react';
import logo from 'media/logo.png';

// routing
import { Link, NavLink } from 'react-router-dom';
import * as routes from 'constants/routes';

// components
import HamButton from '../HamButton';

// hooks
import useMediaQuery from '../../hooks/useMediaQuery';

// use a boolean, not a toggle.  This prevents odd menu states.
// TODO - use tested hooks file - remove hook from here
const useBoolean = (initialValue) => {
  const [status, setStatus] = useState(initialValue);
  const clicked = () => setStatus(true);
  const notClicked = () => setStatus(false);
  window.scrollTo(0, 0);
  return [status, clicked, notClicked];
};

const NavBarNew = () => {
  const small = useMediaQuery(`screen and (max-width: 575px)`);
  const [status, clicked, notClicked] = useBoolean(false);

  if (small) {
    return (
      <>
        <nav className="navbar fixed-top navbar-expand-sm navbar-light bg-light">
          {/* Brand */}
          <Link to="/" className="navbar-brand" onClick={notClicked}>
            <img
              className="d-inline-block align-top"
              src={logo}
              width="30"
              height="30"
              alt=""
            />{' '}
            Verite Data Science
          </Link>

          {/* Burger menu */}
          <HamButton
            type="button"
            toggler={status ? notClicked : clicked}
            clicked={status}
            aria-label="Toggle navigation"
          />
        </nav>

        {status ? (
          <div className="overlay bg-light text-right">
            <NavLink
              className="nav-link"
              activeClassName="active"
              to={routes.ABOUT}
              onClick={notClicked}
            >
              About
            </NavLink>
            <NavLink
              className="nav-link"
              activeClassName="active"
              to={routes.CONTACT}
              onClick={notClicked}
            >
              Contact
            </NavLink>
          </div>
        ) : null}
      </>
    );
  }

  return (
    <nav className="navbar fixed-top navbar-expand-sm navbar-light bg-light">
      <div className="container">
        {/* Brand */}
        <Link to="/" className="navbar-brand">
          <img
            className="d-inline-block align-top"
            src={logo}
            width="30"
            height="30"
            alt=""
          />{' '}
          Verite Data Science
        </Link>

        {/* Links */}
        <div className="navbar-collapse collapse">
          <div className="mx-auto" />
          <NavLink
            className="nav-link"
            activeClassName="active"
            to={routes.ABOUT}
          >
            About
          </NavLink>
          <NavLink
            className="nav-link"
            activeClassName="active"
            to={routes.CONTACT}
          >
            Contact
          </NavLink>
        </div>
      </div>
    </nav>
  );
};

export default NavBarNew;
