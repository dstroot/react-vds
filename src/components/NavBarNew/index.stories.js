import React from 'react';
import { storiesOf } from '@storybook/react';

// components
import NavBar from '../NavBarNew';

// NavBar section
storiesOf('NEXT|NavBarNew', module)
  .addParameters({
    // default for all stories in this book
    info: {
      inline: true,
      header: false,
      text: `
        NavBar provides a navigation component that uses "@reach/router" for links. 
        It overrides some Bootstrap styles so it's "styles.scss" needs to be imported after Bootstrap in "/scss/main.scss".
        `,
    },
  })
  .add('show NavBar', () => <NavBarNew />);
