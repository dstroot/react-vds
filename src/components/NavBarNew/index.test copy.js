import React from 'react';
import { render } from '@testing-library/react';

// component to test
import NavBar from '.';

describe('NavBar', () => {
  it('renders', () => {
    // arange

    // act
    const { asFragment } = render(<NavBar />);

    // assert
    expect(asFragment()).not.toBeNull();
    expect(asFragment()).toMatchSnapshot();
  });
});
