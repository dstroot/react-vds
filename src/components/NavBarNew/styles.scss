//
// Main navbar
//

$height: 75px;

.navbar {
  // box-shadow: 0 2px 40px 0 rgba(0, 0, 0, 0.1);
  border-bottom: 1px solid rgba(200, 200, 200, 1);
  text-transform: uppercase;
  height: $height;
  a {
    text-transform: uppercase;
    font-weight: 400;
    // color: theme-color-level('primary', -8);
    color: gray('700');
    &:hover {
      color: theme-color('primary');
    }
  }
  a.active {
    color: theme-color('primary');
  }
}

.overlay {
  padding-top: $height;
  box-shadow: 0 2px 20px 0 rgba(0, 0, 0, 0.1);

  a {
    font-weight: 400;
    // background: white;
    color: gray('700');
    font-size: 22px;
    padding: 10px;
    padding-right: 20px;
    border-bottom: 1px solid rgba(200, 200, 200, 1);
    &:hover {
      color: theme-color('primary');
    }
  }
  a.active {
    color: theme-color('primary');
  }
}

main {
  margin-top: $height; // same as navbar height
}
