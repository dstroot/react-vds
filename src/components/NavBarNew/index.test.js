// After importing the necessary packages, we can write a test which calls
// the render function provided by React Testing Library, which returns an
// object that we can extract asFragment from. Using this we can use the
// built-in functionality from Jest to perform a snapshot test.

import React from 'react';
import { render, fireEvent, screen, waitFor } from '@testing-library/react';
import { renderHook, act } from '@testing-library/react-hooks';

// component to test
import NavBarNew from '.';

describe('NavBarNew', () => {
  it('renders', () => {
    // arrange

    // act
    const { asFragment } = render(<NavBarNew />);

    // assert
    expect(asFragment()).not.toBeNull();
    expect(asFragment()).toMatchSnapshot();
  });

  it('it should be at the top of the page', () => {
    // arrange
    const { container } = render(<NavBarNew />);
    const nav = container.querySelector('nav');

    //act

    //assert

    // Verify its class includes "fixed top"
    expect(nav.className).toContain('fixed-top');
  });

  it('it should render small', async () => {
    // arrange

    expect(window.innerWidth).toBe(1024);
    expect(window.innerHeight).toBe(768);

    act(() => {
      // Change the viewport to 500px.
      window.innerWidth = 319;
      window.innerHeight = 500;

      // Trigger the window resize event.
      // window.dispatchEvent(new Event('resize'))
      fireEvent(window, new Event('resize'));
    });

    // assert]
    expect(window.innerWidth).toBe(319);
    expect(window.innerHeight).toBe(500);

    const { asFragment } = render(<NavBarNew />);
    // await waitFor(() => expect(asFragment()).not.toBeNull());
    // await waitFor(() => expect(asFragment()).toMatchSnapshot());
    expect(asFragment()).not.toBeNull();
    expect(asFragment()).toMatchSnapshot();
  });
});
