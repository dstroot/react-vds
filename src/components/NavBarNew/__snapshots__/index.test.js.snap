// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarNew it should render small 1`] = `
<DocumentFragment>
  <nav
    class="navbar fixed-top navbar-expand-sm navbar-light bg-light"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        <img
          alt=""
          class="d-inline-block align-top"
          height="30"
          src="logo.png"
          width="30"
        />
         Verite Data Science
      </a>
      <div
        class="navbar-collapse collapse"
      >
        <div
          class="mx-auto"
        />
        <a
          class="nav-link"
          href="/about"
        >
          About
        </a>
        <a
          class="nav-link"
          href="/contact"
        >
          Contact
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`NavBarNew renders 1`] = `
<DocumentFragment>
  <nav
    class="navbar fixed-top navbar-expand-sm navbar-light bg-light"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        <img
          alt=""
          class="d-inline-block align-top"
          height="30"
          src="logo.png"
          width="30"
        />
         Verite Data Science
      </a>
      <div
        class="navbar-collapse collapse"
      >
        <div
          class="mx-auto"
        />
        <a
          class="nav-link"
          href="/about"
        >
          About
        </a>
        <a
          class="nav-link"
          href="/contact"
        >
          Contact
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
